Source/WebKit/WebKitSwift/LinearMediaKit/LinearMediaTypes.swift

```diff
@@ -0,0 +1,188 @@
+        case .scaleToFill:
+            self = .scaleToFill
+        @unknown default:
+            fatalError()
+        }
+    }
+
+    var contentMode: ContentMode? {
+        switch self {
+        case .none:
+            return nil
+        case .scaleAspectFit:
+            return .scaleAspectFit
+        case .scaleAspectFill:
+            return .scaleAspectFill
+        case .scaleToFill:
+            return .scaleToFill
+        @unknown default:
+            fatalError()
+        }
+    }
+
+    static var `default`: WKSLinearMediaContentMode = .init(.default)
+}
+
+extension WKSLinearMediaContentType {
+    var contentType: ContentType? {
+        switch self {
+        case .none:
+            return nil
+        case .immersive:
+            return .immersive
+        case .spatial:
+            return .spatial
+        case .planar:
+            return .planar
+        case .audioOnly:
+            return .audioOnly
+        @unknown default:
+            fatalError()
+        }
+    }
+}
+
+extension WKSLinearMediaPresentationMode {
+    var presentationMode: PresentationMode? {
+        switch self {
+        case .none:
+            return nil
+        case .inline:
+            return .inline
+        case .fullscreen:
+            return .fullscreen
+        case .fullscreenFromInline:
+            return .fullscreenFromInline
+        case .pip:
+            return .pip
+        @unknown default:
+            fatalError()
+        }
+    }
+}
+
+extension WKSLinearMediaViewingMode {
+    init(_ viewingMode: ViewingMode?) {
+        guard let viewingMode else {
+            self = .none
+            return
+        }
+
+        switch viewingMode {
+        case .mono:
+            self = .mono
+        case .stereo:
+            self = .stereo
+        case .immersive:
+            self = .immersive
+        case .spatial:
+            self = .spatial
+        @unknown default:
+            fatalError()
+        }
+    }
+    
+    var viewingMode: ViewingMode? {
+        switch self {
+        case .none:
+            return nil
+        case .mono:
+            return ViewingMode.mono
+        case .stereo:
+            return ViewingMode.stereo
+        case .immersive:
+            return ViewingMode.immersive
+        case .spatial:
+            return ViewingMode.spatial
+        @unknown default:
+            fatalError()
+        }
+    }
+}
+
+extension WKSLinearMediaFullscreenBehaviors {
+    init(_ fullscreenBehaviors: FullscreenBehaviors) {
+        self = .init(rawValue: fullscreenBehaviors.rawValue)
+    }
+
+    var fullscreenBehaviors: FullscreenBehaviors {
+        .init(rawValue: self.rawValue)
+    }
+}
+
+extension WKSLinearMediaTimeRange {
+    var closedRange: ClosedRange<TimeInterval> {
+        return lowerBound...upperBound
+    }
+
+    var range: Range<TimeInterval> {
+        return lowerBound..<upperBound
+    }
+}
+
+extension WKSLinearMediaTrack: @retroactive Track {
```

Comment:
what the heck does retroactive do

Reply:
It silences this warning:
```
Extension declares a conformance of imported type 'WKSLinearMediaTrack' to imported protocol 'Track'; this will not behave correctly if the owners of 'WebKitSwift' introduce this conformance in the future
```

... which is a silly warning, because we are the actual owners of the WebKitSwift clang module (i.e. the stuff in WebKitSwift.h).

Reply:
oh wow, cool